test/integration/test_kongplugininstallation.go

```diff
@@ -378,10 +381,18 @@ func verifyCustomPlugins(t *testing.T, ip string, expectedHeaders ...http.Header
 		}
 		for _, h := range expectedHeaders {
 			for k, v := range h {
-				assert.Equal(c, v, resp.Header.Values(k))
+				actualV := resp.Header.Values(k)
```

Comment:
WDYT about something like that as body of `require.EventuallyWithT(t, func(c *assert.CollectT) {...` to not spam with logs?

```go
	require.EventuallyWithT(t, func(c *assert.CollectT) {
		readBody := func(resp *http.Response) string {
			var b bytes.Buffer
			_, err := b.ReadFrom(resp.Body)
			assert.NoError(c, err, "failed to read response body")
			return b.String()
		}
		resp, err := httpClient.Get(fmt.Sprintf("http://%s/test", ip))
		if !assert.NoError(c, err) {
			return
		}
		defer resp.Body.Close()
		if !assert.Equal(c, http.StatusOK, resp.StatusCode, "unexpected status code, body: %s", readBody(resp)) {
			return
		}
		for _, h := range expectedHeaders {
			for k, v := range h {
				assert.Equal(c, v, resp.Header.Values(k), "unexpected header, body: %s", readBody(resp))
			}
		}
	}, time.Minute, 250*time.Millisecond)
```

Reply:
👍 
d8ffa83e581e7e5a38bb7ae33c91fb43b43a3b72